src/include/mpir_request.h

```diff
@@ -236,6 +225,11 @@ struct MPIR_Request {
         } rma;                  /* kind : MPIR_REQUEST_KIND__RMA */
     } u;
 
+#if defined HAVE_DEBUGGER_SUPPORT
+    struct MPIR_Debugq *posted;
+    struct MPIR_Debugq *unexp;
+#endif                          /* HAVE_DEBUGGER_SUPPORT */
```

Comment:
I guess this addressed my previous comment.

Where is the sendq pointer?

Reply:
I avoided naming the pointers send/recv_foo because of the unique situation in ch4/self where a request can be both posted send and unexpected recv. So "posted" is basically either the posted send or recv queue. Open to suggestions to make it clearer.

Reply:
I see. I would suggest using separate fields for clearer reading.

Reply:
Yeah, it maps to three debug queues straightforwardly.

Reply:
It might also be possible to do this, but maybe its more confusing...
```c
#if defined HAVE_DEBUGGER_SUPPORT
    union {
        struct MPIR_Debugq *send;
        struct MPIR_Debugq *recv;
    } u;
    struct MPIR_Debugq *unexp;
#endif
```

Reply:
It's only turned on with debugger support, so I would rather flatten it and have a easier reading than save one extra field.

Reply:
Done.